src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/ForceMergeAction.kt

```diff
@@ -0,0 +1,78 @@
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.forcemerge.AttemptSetReadOnlyStep
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.forcemerge.WaitForForceMergeStep
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+
+class ForceMergeAction(
+    clusterService: ClusterService,
+    client: Client,
+    managedIndexMetaData: ManagedIndexMetaData,
+    config: ForceMergeActionConfig
+) : Action(ActionType.FORCE_MERGE, config, managedIndexMetaData) {
+
+    private val attemptSetReadOnlyStep = AttemptSetReadOnlyStep(clusterService, client, config, managedIndexMetaData)
+    private val attemptCallForceMergeStep = AttemptCallForceMergeStep(clusterService, client, config, managedIndexMetaData)
+    private val waitForForceMergeStep = WaitForForceMergeStep(clusterService, client, config, managedIndexMetaData)
+    private val attemptRevertReadOnlyStep = AttemptRevertReadOnlyStep(clusterService, client, config, managedIndexMetaData)
+
+    // Using a LinkedHashMap here to maintain order of steps for getSteps() while providing a convenient way to
+    // get the current Step object using the current step's name in getStepToExecute()
+    private val stepNameToStep: LinkedHashMap<String, Step> = linkedMapOf(
+        AttemptSetReadOnlyStep.name to attemptSetReadOnlyStep,
+        AttemptCallForceMergeStep.name to attemptCallForceMergeStep,
+        WaitForForceMergeStep.name to waitForForceMergeStep,
+        AttemptRevertReadOnlyStep.name to attemptRevertReadOnlyStep
+    )
+
+    override fun getSteps(): List<Step> = stepNameToStep.values.toList()
+
+    @Suppress("ReturnCount")
+    override fun getStepToExecute(): Step {
+        // If stepMetaData is null, return the first step in ForceMergeAction
+        val stepMetaData = managedIndexMetaData.stepMetaData ?: return attemptSetReadOnlyStep
+        val currentStep = stepMetaData.name
+
+        // If the current step is not from this action (assumed to be from the previous action in the policy), return
+        // the first step in ForceMergeAction
+        if (!stepNameToStep.containsKey(currentStep)) return attemptSetReadOnlyStep
+
+        val currentStepStatus = stepMetaData.stepStatus
+
+        // If the current step has completed, return the next step
+        if (currentStepStatus == Step.StepStatus.COMPLETED) {
+            return when (currentStep) {
+                AttemptSetReadOnlyStep.name -> attemptCallForceMergeStep
+                AttemptCallForceMergeStep.name -> waitForForceMergeStep
+                WaitForForceMergeStep.name -> attemptRevertReadOnlyStep
+                // Shouldn't hit this case but including it so that the when expression is exhaustive
+                else -> throw IllegalArgumentException("Unexpected step: [$currentStep] when getting step to execute for ForceMergeAction")
```

Comment:
Are you catching this somewhere? Otherwise this is throwing an uncaught exception in the middle the runner.

Reply:
Actually reverted this to the old check for a couple reasons:
1. As you mentioned, to not possibly throw an exception in the middle of runner
2. The additional ```if``` check we have before this ```when``` statement now should prevent any cases where an unexpected ```currentStep``` is getting evaluated here. If for some reason we do reach this ```else``` it would mean that we are on the last step and the runner did not properly evaluate we were on the last step and tried to get a step to execute for some reason. If this happens, returning the currentStep (last step in this case) would be the same behavior we currently have for our other Actions